### PR TITLE
feat: added a transformation library to localize apple device models

### PR DESCRIPTION
**Pull request by desusai7**
**Added a transformation library to localize apple device models** 

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update
## Checklist:
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added unit tests for the code
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: libraries/localizeAppleDeviceModel/v1/code.js [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+    "iPhone9,3": "iPhone 7",
+    "iPhone9,4": "iPhone 7 Plus",
+    "iPhone10,1": "iPhone 8",
+    "iPhone10,4": "iPhone 8",
+    "iPhone10,2": "iPhone 8 Plus",
+    "iPhone10,5": "iPhone 8 Plus",
+    "iPhone10,3": "iPhone X",
+    "iPhone10,6": "iPhone X",
+    "iPhone11,2": "iPhone XS",
+    "iPhone11,6": "iPhone XS Max",
```
Comment (ChryssaAliferi) [inline libraries/localizeAppleDeviceModel/v1/code.js]: I found also an "iPhone11,4": "iPhone XS Max".

Comment (ChryssaAliferi) [reply libraries/localizeAppleDeviceModel/v1/code.js]: Would you like to cross check?

**Review by ChryssaAliferi — COMMENTED** [review]
Also I was thinking about the Apple TV devices. Should we include them in this list?

File: libraries/localizeAppleDeviceModel/v1/code.js (hunk 1 above)
Comment (desusai7) [reply libraries/localizeAppleDeviceModel/v1/code.js]: I've checked it and found some references where `iPhone11,4` and `iPhone11,6` were tagged as iPhone XS Max. Adding `iPhone11,4`:`iPhone XS Max` to the mappings as well.

**Comment by desusai7** [comment]
@ChryssaAliferi, added apple tv support and few minor improvements, please take a look again

**Review by ChryssaAliferi — APPROVED** [review]

**Comment by Jayachand** [comment]
Can you give sample transformation code here?

**Review by Jayachand — APPROVED** [review]

File: libraries/localizeAppleDeviceModel/v1/code.js [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+  return false;
+}
+
+export function localizeDeviceModel(event) {
+  let devicePlatform = event?.context?.os?.name;
+  let currentDeviceModel = event?.context?.device?.model;
+  if (devicePlatform && isAppleFamily(devicePlatform) && currentDeviceModel) {
+    event.context.device.model = getLocalizedDeviceModel(currentDeviceModel);
+  }
+  return event;
```
Comment (Jayachand) [inline libraries/localizeAppleDeviceModel/v1/code.js]: nit: prefers to return model rather than event

Comment (desusai7) [reply libraries/localizeAppleDeviceModel/v1/code.js]: updated the library to return model instead of event

**Pull request merged**